### PR TITLE
Search assets max limit

### PR DESCRIPTION
**Pull request by itssapir**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Fixes: [CRTX-250550](https://jira-dc.paloaltonetworks.com/browse/CRTX-250550)

## Description
Fix search assets not mentioning the default limit on the output and failing for large limits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Core/Integrations/CortexPlatformCore/CortexPlatformCore.py [hunk 1]
```diff
@@ -2303,6 +2304,12 @@ def search_assets_command(client: Client, args):
     filter.add_field(ASSET_FIELDS["asset_types"], FilterType.CONTAINS, asset_types)
 
     page_size: int = arg_to_number(args.get("page_size", SEARCH_ASSETS_DEFAULT_LIMIT))  # type: ignore[assignment]
+    if page_size > SEARCH_ASSETS_MAX_LIMIT:
```
Comment (marketplace-ai-reviewer) [inline Packs/Core/Integrations/CortexPlatformCore/CortexPlatformCore.py:2309]: - Please add unit tests to verify the new `page_size` logic, specifically covering cases where `page_size` is `0` and when it exceeds `SEARCH_ASSETS_MAX_LIMIT`.
- Handle potential `None` value from `arg_to_number` to prevent `TypeError`.

File: Packs/Core/Integrations/CortexPlatformCore/CortexPlatformCore.yml [hunk 2]
```diff
@@ -1457,7 +1457,7 @@ script:
       description: Updates the properties of an issue. This command does not provide an explicit indication of success.
     - arguments:
       - name: page_size
-        description: The number of assets to return per page. Default is 100.
+        description: The number of assets to return per page. Default is 100. Max is 5000.
```
Comment (marketplace-ai-reviewer) [inline Packs/Core/Integrations/CortexPlatformCore/CortexPlatformCore.yml]: Please move the default value to the `defaultValue` field and remove it from the description.

```suggestion
        description: The number of assets to return per page. Max is 5000.
        defaultValue: "100"
```

File: Packs/Core/ReleaseNotes/3_5_52.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Cortex Platform - Core
+
+- Added a max page size to the ***core-search-assets*** command to avoid crashes.
```
Comment (marketplace-ai-reviewer) [inline Packs/Core/ReleaseNotes/3_5_52.md]: The release note should style the argument name using italics (`*page_size*`) and clearly state the maximum limit for better user clarity. Additionally, mentioning internal reasons like "to avoid crashes" is generally discouraged as it is developer-facing information.

Consider updating the release note to:
```suggestion
- Added a maximum limit of 5000 to the *page_size* argument in the ***core-search-assets*** command.
```

Please refer to the [General](https://xsoar.pan.dev/docs/documentation/release-notes#general) guidelines for release notes. You can also auto-generate release notes using `demisto-sdk update-release-notes`.

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the Cortex Platform Core integration! I've reviewed the PR and left a few notes to help polish the changes. 

Primarily, please add unit tests for the new `page_size` boundaries and ensure potential `None` values are handled safely in the code. You'll also see a few minor adjustments needed for the YAML structure, README, and Release Notes to align with our documentation and formatting standards. 

Let me know if you have any questions!

Additionally, please address the following file-level notes:
- `Packs/Core/Integrations/CortexPlatformCore/CortexPlatformCore.py`: Missing unit tests for new boundary conditions.
- `Packs/Core/Integrations/CortexPlatformCore/CortexPlatformCore.yml`: - The first key defined at the root level of the `.yml` file must always be `name` or `display`. Currently, it starts with `category`.
- Please ensure the `README.md` file is updated to reflect the new maximum limit for the `page_size` argument in the `core-search-assets` command.

 @itssapir please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Review by omerKarkKatz — APPROVED** [review]
